LuaValue.cpp

```diff
@@ -143,11 +140,13 @@ int LuaVal::lua_set(lua_State* L) {
     return 0;
 }
 
-std::string LuaVal::tostring(void* ptr)
+std::string LuaVal::to_string_map(MapType const* ptr)
 {
-    char arr[128];
-    snprintf(arr, 128, "LuaVal: %p", ptr);
-    return arr;
+    std::string out = "[\n";
+    for (std::pair<const LuaVal, LuaVal> const& pair : *ptr)
+        out += "  { key: " + pair.first.to_string() + ", value: " + pair.second.to_string() + " },\n";
+    out += ']';
+    return out;
```

Comment:
This is not what I would personally expect tostring to return on a map or table in in lua necessarily.
While primitives print the value itself, for more complex types normally the type name and pointer is printed.
For example `table: 0x55d4aaf5dd40`, `userdata: 0x8064d48`

Reply:
True, but a pointer is of questionable utility when debugging (which is what you would use a tostring for on a complex object)

Reply:
We could also add a luaval:Serialize() method to get the stringified version of the contents.